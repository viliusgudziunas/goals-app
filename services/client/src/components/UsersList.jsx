import React from 'react';

const UsersList = ({ users }) => {
  return (
    <div className="list-group list-group-flush">
      {users.map(user => {
        return (
          <div className="list-group-item" key={user.id}>
            {user.email}
          </div>
        );
      })}
    </div>
  );
};

export default UsersList;
