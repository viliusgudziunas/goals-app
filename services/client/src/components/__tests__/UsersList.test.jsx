import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import UsersList from '../UsersList';

const users = [
  { active: true, email: 'test@test.com', id: 1 },
  { active: true, email: 'test2@test.com', id: 2 }
];

describe('UsersList', () => {
  it('renders a snapshot properly', () => {
    const tree = renderer.create(<UsersList users={users} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  const wrapper = shallow(<UsersList users={users} />);
  it('renders properly', () => {
    const element = wrapper.find('.list-group-item');

    expect(element.length).toBe(2);
    expect(element.get(0).props.children).toBe('test@test.com');
    expect(element.get(1).props.children).toBe('test2@test.com');
  });
});
