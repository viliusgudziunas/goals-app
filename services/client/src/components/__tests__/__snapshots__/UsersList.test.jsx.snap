// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList /> should match the snapshot 1`] = `"<div class=\\"row\\"><div class=\\"col\\"><div class=\\"row list-group list-group-flush\\"><div class=\\"list-group-item\\">test@test.com</div><div class=\\"list-group-item\\">test2@test.com</div></div></div></div>"`;
